libvast/vast/detail/concepts.hpp

```diff
@@ -41,4 +56,26 @@ concept unsigned_integral = integral<T> && std::is_unsigned_v<T>;
 template <class T>
 concept signed_integral = integral<T> && std::is_signed_v<T>;
 
+struct any_callable {
+  template <class... Ts>
+  void operator()(Ts&&...);
+};
+
+/// Inspectables
+template <class T>
+concept is_inspectable = requires(detail::any_callable& i, T& x) {
```

Comment:
You can skip the `detail` namespace qualification as we're already in the `detail` namespace.

Nit regarding naming: thoughts on `inspectable` instead of `is_inspectable`?